prow/cmd/horologium/main.go

```diff
@@ -180,39 +181,34 @@ func sync(prowJobClient ctrlruntimeclient.Client, cfg *config.Config, cr cronCli
 				}
 				shouldTrigger = now.Sub(intervalRef) > intervalDuration
 			}
-			logger = logger.WithField("should-trigger", shouldTrigger)
-			if !previousFound || shouldTrigger {
-				prowJob := pjutil.NewProwJob(pjutil.PeriodicSpec(p), p.Labels, p.Annotations)
-				prowJob.Namespace = cfg.ProwJobNamespace
-				logger.WithFields(pjutil.ProwJobFields(&prowJob)).Info("Triggering new run of interval periodic.")
-				if err := prowJobClient.Create(context.TODO(), &prowJob); err != nil {
-					errs = append(errs, err)
-				}
-			}
-		} else if cronTriggers.Has(p.Name) {
-			shouldTrigger := j.Complete()
-			logger = logger.WithField("should-trigger", shouldTrigger)
-			if !previousFound || shouldTrigger {
-				prowJob := pjutil.NewProwJob(pjutil.PeriodicSpec(p), p.Labels, p.Annotations)
-				prowJob.Namespace = cfg.ProwJobNamespace
-				logger.WithFields(pjutil.ProwJobFields(&prowJob)).Info("Triggering new run of cron periodic.")
-				if err := prowJobClient.Create(context.TODO(), &prowJob); err != nil {
-					errs = append(errs, err)
-				}
-			} else {
+		case cronTriggers.Has(p.Name):
+			shouldTrigger = j.Complete()
+		default:
+			if !cronTriggers.Has(p.Name) {
 				logger.WithFields(logrus.Fields{
 					"previous-found": previousFound,
 					"should-trigger": shouldTrigger,
 					"name":           p.Name,
 					"job":            p.JobBase.Name,
-				}).Info("skipping cron periodic")
+				}).Info("Skipping cron periodic")
+			}
+			continue
+		}
+		if shouldTrigger {
```

Comment:
We don't need this conditional. If `should-trigger` is false that seems like something important to know while debugging.